apps/explorer/app/tx/[id]/page.tsx

```diff
@@ -23,21 +23,40 @@ export const revalidate = 0
 
 export default async function Page({ params }) {
   const id = params?.id as string
-  const [transaction, error] = await to(
-    siaCentral.transaction({
-      params: {
-        id,
-      },
-    })
-  )
 
-  if (error) {
-    throw error
-  }
+  // Make all non-dependent requests together.
+  const [
+    [transaction, transactionError],
+    [transactionChainIndices, transactionChainIndicesError],
+    [currentTip, currentTipError],
+  ] = await Promise.all([
+    to(explored.transactionByID({ params: { id } })),
+    to(explored.transactionChainIndices({ params: { id } })),
+    to(explored.consensusTip()),
+  ])
+
+  if (transactionError) throw transactionError
+  if (transactionChainIndicesError) throw transactionChainIndicesError
+  if (currentTipError) throw currentTipError
+  if (!transaction || !transactionChainIndices || !currentTip) return notFound()
 
-  if (!transaction?.transaction) {
-    return notFound()
-  }
+  // Use the first chainIndex from the above call to get our parent block.
+  const [parentBlock, parentBlockError] = await to(
+    explored.blockByID({ params: { id: transactionChainIndices[0].id } })
+  )
 
-  return <Transaction transaction={transaction.transaction} />
+  if (parentBlockError) throw parentBlockError
+  if (!parentBlock) return notFound()
+
+  return (
+    <Transaction
+      transaction={transaction}
+      transactionHeaderData={{
+        id: stripPrefix(transaction.id),
+        blockHeight: transactionChainIndices[0].height,
+        confirmations: currentTip.height - transactionChainIndices[0].height,
+        timestamp: parentBlock.timestamp,
+      }}
+    />
+  )
```

Comment:
Nice with the parallelizing where possible and also processing and passing minimal required data from the server. 👍🏼